src/aws-cpp-sdk-core/source/utils/crypto/factory/Factories.cpp

```diff
@@ -45,6 +46,12 @@ static std::shared_ptr<HashFactory>& GetCRC32CFactory()
     return s_CRC32CFactory;
 }
 
+static std::shared_ptr<HashFactory>& GetCRC64Factory()
+{
+    static std::shared_ptr<HashFactory> s_CRC64Factory(nullptr);
```

Comment:
how does it work with our custom memory management?

Reply:
its gets assigned when we call its factory, it follows the same call path as CRC32 does. i.e.

[create accessor for static factory](https://github.com/aws/aws-sdk-cpp/blob/main/src/aws-cpp-sdk-core/source/utils/crypto/factory/Factories.cpp#L36-L40)
```cpp
static std::shared_ptr<HashFactory>& GetCRC32Factory()
{
    static std::shared_ptr<HashFactory> s_CRC32Factory(nullptr);
    return s_CRC32Factory;
}
```

then is assigned during the [init crypto call](https://github.com/aws/aws-sdk-cpp/blob/main/src/aws-cpp-sdk-core/source/utils/crypto/factory/Factories.cpp#L549-L552)
```cpp
if(!GetCRC32Factory())
{
  GetCRC32Factory() = Aws::MakeShared<DefaultCRC32Factory>(s_allocationTag);
}
```

Reply:
OMG, okay.

Reply:
yeah changing this pattern and avoiding static state would be a larger change than this, but I agree something that should be done as part of some future "eliminate static state" task